zerver/views/streams.py

```diff
@@ -772,6 +773,8 @@ def get_topics_backend(
 
     if is_web_public_query:
         realm = get_valid_realm_from_request(request)
+        if not realm.allow_web_public_streams_access():
+            raise MissingAuthenticationError()
```

Comment:
I think probably the right model is to have this check occur inside `access_web_public_stream`; generally for authorization things, one wants to rely on common code as much as possible.

Reply:
Updated.